### DISCUSSION TITLE
Drag mouse pointer furthur instead of clicking the icons.

### DISCUSSION THREAD

**Discussion by ranjith210** [Schneegans/Fly-Pie#84] [Ideas & Feature Discussions]
An alternate option to drag the mouse pointer for selecting items (sub-menu).

**Comment by Schneegans** [comment]
Hi there! I fear that I do not fully understand what you would like to have... Could you try to explain it more in detail? Currently we have these interaction modes:
* **Point-and-Click:** Select items by clicking on them or anywhere in their wedge.
* **Marking-Mode:** Select items by drawing gestures. To do this, click the center item (or anywhere else) and drag your mouse with the pressed button over the item you want to select.

I think what you suggest may be a marking mode without pressing the mouse button...? I already thought about something like this to improve touchpad support. Dragging on a touchpad is actually pretty awkward...

**Comment by Schneegans** [comment]
I just implemented something I called "Turbo-Mode": You can now also "draw" gestures as long as a modifier key, such as <kbd>Ctrl</kbd>, <kbd>Shift</kbd>, or <kbd>Alt</kbd> is held down without having to press your mouse button! This is especially useful when you opened the menu with a shortcut involving such a modifier key. You can just keep it pressed and move the pointer with the mouse or your touch-pad!

You may test this new feature if you [clone the latest develop branch from GitHub](https://github.com/Schneegans/Fly-Pie/blob/develop/docs/installation.md#c-cloning-the-latest-version-with-git)!

**Comment by ykhurshid** [comment]
I was actually about to create an unnecessary issue along the lines that I believe the OP is thinking. 

It would be nice to have the option of having turbo mode enabled without the need for a modifier key. I use easystroke to invoke most of the fly-pie menus I have and I generally use the side mouse buttons for these. It's probably both more ergonomic and faster to be able to choose a slice without needing to hold down a mouse or keyboard button. There can be a customized threshold for how far you need to drag the cursor out for a menu option to be chosen. 

We can navigate down a pie-menu's hierarchy by simply pointing the cursor back on the sub-menu's source icon like how turbo mode works already. To cancel out of the entire menu, we can just use the secondary mouse button.

**Reply by Schneegans** [reply]
I agree that this may be even faster for experts! However, the cool thing about the three interaction modes we currently have is that they are not-exclusive, so you can change between point-and-click mode, marking mode and turbo mode whenever you want. How would you suggest to enter this click-less turbo mode (lets call it "hyper mode" :smile: )? Would it be a per-menu setting, like the "Open in screen center"-option? Or a global option?

**Reply by ykhurshid** [reply]
That's a good question. I would always prefer more customizability provided it does not interfere with usability so It should probably be a per-menu setting as well as a global setting for people who have mastered the other methods.

**Comment by Schneegans** [comment]
I just implemented an experimental "Hover Mode". If you checkout `develop`, you can enable it in the advanced settings of the preferences dialog. With a bit of practice, it's quite usable, I think. What's your opinion?

**Reply by ykhurshid** [reply]
I'm testing it now and after increasing the minimum stroke length from 150 to 200, it feels fantastic. It's especially made certain pie menu applications such as window management feel effortless now and preferable over having to drag the pointer all the way over to the top (left or right depending on your gnome theme) corner of the window.

I'll let you know if I run into any usability problems down the line. Thank you for the great work!
